### ISSUE TITLE
Consensus failure on genesis initial height other than 1

### ISSUE THREAD

**Issue by dckc** [Agoric/agoric-sdk#2899]
## Describe the bug

Symptoms:

```
2:01AM ERR CONSENSUS FAILURE!!! err="Error: Unimplemented reset state from 295227 to 295226" module=consensus stack="goroutine 1195 [running]:\nruntime/debug.Stack(0xc004697f48, 0x7fe153121c20, 0xc00400b5d0)\n\t/usr/local/go/src/runtime/debug/stack.go:24
```

Diagnosis:

Our chain runs the economy bootstrap pieces (#2748) before finishing genesis, then goes on to connect to the network and process the first block (#2864).  When initial_height was "1", this worked correctly and computed the expected block height, but that calculation did not work for initial_height greater than 1. The incorrect calculation caused a key assertion to fail, which crashed the node.

When testing the genesis that was published for phase 2, we tested for the case of a starting time in the future but didn't also test an initial height other than 1. We weren't able to test the _actual_ phase 2 genesis block because we didn't have the corresponding private keys; it required a decentralized start.

After isolating the error, we were able to reproduce it.

**Comment by dckc** [comment]
fixed in #2895

cc @warner @dtribble @michaelfig (see symptoms, diagnosis in the issue description)

**Comment by dtribble** [comment]
Additional detail:

The chain startup runs the economy bootstrap pieces before finishing genesis, then goes on to connect to the network and process the first block.  When `initial_height` was 1, this worked correctly and computed the expected block height, but that calculation did not work for `initial_height` greater than 1. The incorrect calculation caused a key assertion to fail, which halted the node.
